numpy/ma/core.py

```diff
@@ -7400,22 +7401,195 @@ def mask_rowcols(a, axis=None):
+    >>> a
+    masked_array(
+      data=[[[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]],
+
+            [[0, 0, 0],
+             [0, --, 0],
+             [0, 0, 0]],
+
+            [[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]],
+      mask=[[[False, False, False],
+             [False, False, False],
+             [False, False, False]],
+
+            [[False, False, False],
+             [False,  True, False],
+             [False, False, False]],
+
+            [[False, False, False],
+             [False, False, False],
+             [False, False, False]]],
+      fill_value=1)
+    >>> ma.mask_extend_axis(a, axis=1)
+    masked_array(
+      data=[[[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]],
+
+            [[0, --, 0],
+             [0, --, 0],
+             [0, --, 0]],
+
+            [[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]]],
+      mask=[[[False, False, False],
+             [False, False, False],
+             [False, False, False]],
+
+            [[False,  True, False],
+             [False,  True, False],
+             [False,  True, False]],
+
+            [[False, False, False],
+             [False, False, False],
+             [False, False, False]]],
+      fill_value=1)
+    >>> ma.mask_extend_axis(a, axis=(1, 2))
+    masked_array(
+      data=[[[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]],
+
+            [[0, --, 0],
+             [--, --, --],
+             [0, --, 0]],
+
+            [[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]]],
+      mask=[[[False, False, False],
+             [False, False, False],
+             [False, False, False]],
+
+            [[False,  True, False],
+             [ True,  True,  True],
+             [False,  True, False]],
+
+            [[False, False, False],
+             [False, False, False],
+             [False, False, False]]],
+      fill_value=1)
+
+    Note that calling this function twice is not equivalent to passing axes
+    simultaneously:
+
+    >>> ma.mask_extend_axis(ma.mask_extend_axis(a, axis=1), axis=2)
+    masked_array(
+      data=[[[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]],
+
+            [[--, --, --],
+             [--, --, --],
+             [--, --, --]],
+
+            [[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]]],
+      mask=[[[False, False, False],
+             [False, False, False],
+             [False, False, False]],
+
+            [[ True,  True,  True],
+             [ True,  True,  True],
+             [ True,  True,  True]],
+
+            [[False, False, False],
+             [False, False, False],
+             [False, False, False]]],
+      fill_value=1)
+    """
+    a = array(a, subok=False)
+    if axis is None:
+        axis = range(a.ndim)
+    else:
+        axis = normalize_axis_tuple(axis, a.ndim)
     m = getmask(a)
     # Nothing is masked: return a
-    if m is nomask or not m.any():
```

Comment:
`m.any` is basically computed by the `logical_or.reduce` below, and probably isn't worthwhile